test_conformance/basic/test_preprocessors.cpp

```diff
@@ -250,33 +232,28 @@ int test_kernel_preprocessor_macros(cl_device_id deviceID, cl_context context, c
         return -1;
     }
 
-    // The OpenCL C version reported by the macro reports the OpenCL C supported by the compiler for this OpenCL device.
-    char cVersionBuffer[ 128 ];
-    error = clGetDeviceInfo( deviceID, CL_DEVICE_OPENCL_C_VERSION, sizeof( cVersionBuffer ), cVersionBuffer, NULL );
-    test_error( error, "Unable to get device's OpenCL C version to validate against" );
+    // The OpenCL C version reported by the macro reports the OpenCL C version
+    // specified to the compiler. We need to see whether it is supported.
 
-    // We need to parse to get the version number to compare against
-    for( p1 = cVersionBuffer; ( *p1 != 0 ) && !isdigit( *p1 ); p1++ )
-        ;
-    for( p2 = p1; ( *p2 != 0 ) && ( *p2 != '.' ); p2++ )
-        ;
-    for( p3 = p2; ( *p3 != 0 ) && ( *p3 != ' ' ); p3++ )
-        ;
-
-    if( p2 == p3 )
+    if ((results[3] < 100)
+        || (!device_supports_cl_c_version(deviceID,
+                                          Version{ results[3] / 10 })))
```

Comment:
This is fairly minor, but I'd consider decoding the result value completely into major and minor versions and using the existing constructor vs. defining a new constructor that works on a combined value.  This seems safer than relying on the current method of converting between a Version and an integer.

Reply:
Sounds reasonable to me!